p2p/src/peer_manager/tests/connections.rs

```diff
@@ -1065,19 +1068,326 @@ where
+        ping_check_period: Default::default(),
+        ping_timeout: Default::default(),
+        peer_handshake_timeout: Default::default(),
+        max_clock_diff: Default::default(),
+        node_type: Default::default(),
+        user_agent: mintlayer_core_user_agent(),
+        sync_stalling_timeout: Default::default(),
+        peer_manager_config: Default::default(),
+        protocol_config: Default::default(),
+    });
+    let (peer_mgr_event_sender2, _shutdown_sender, _subscribers_sender) =
+        run_peer_manager::<DefaultNetworkingService<MpscChannelTransport>>(
+            make_transport_with_local_addr_in_group(2),
+            TestTransportChannel::make_address(),
+            Arc::clone(&chain_config),
+            p2p_config_2,
+            time_getter.get_time_getter(),
+        )
+        .await;
+
+    // Start the third peer manager and let it know about the first peer using reserved
+    let p2p_config_3 = Arc::new(P2pConfig {
+        reserved_nodes,
+        allow_discover_private_ips: true.into(),
+
+        bind_addresses: Default::default(),
+        socks5_proxy: None,
+        disable_noise: Default::default(),
+        boot_nodes: Default::default(),
+        whitelisted_addresses: Default::default(),
+        ban_threshold: Default::default(),
+        ban_duration: Default::default(),
+        outbound_connection_timeout: Default::default(),
+        ping_check_period: Default::default(),
+        ping_timeout: Default::default(),
+        peer_handshake_timeout: Default::default(),
+        max_clock_diff: Default::default(),
+        node_type: Default::default(),
+        user_agent: mintlayer_core_user_agent(),
+        sync_stalling_timeout: Default::default(),
+        peer_manager_config: Default::default(),
+        protocol_config: Default::default(),
+    });
+    let (peer_mgr_event_sender3, _shutdown_sender, _subscribers_sender) =
+        run_peer_manager::<DefaultNetworkingService<MpscChannelTransport>>(
+            make_transport_with_local_addr_in_group(2),
+            TestTransportChannel::make_address(),
+            Arc::clone(&chain_config),
+            p2p_config_3,
+            time_getter.get_time_getter(),
+        )
+        .await;
+
+    let started_at = Instant::now();
+
+    // All peers should discover each other
+    loop {
+        let connected_peers = tokio::join!(
+            get_connected_peers(&peer_mgr_event_sender1),
+            get_connected_peers(&peer_mgr_event_sender2),
+            get_connected_peers(&peer_mgr_event_sender3)
+        );
+        let counts = [connected_peers.0.len(), connected_peers.1.len(), connected_peers.2.len()];
+
+        // There should be:
+        // - 2 outbound and 2 inbound connections to/from reserved peers.
+        // - 3 outbound connections to the discovered addresses
+        //   (each peer can make only one outbound connection because all discovered addresses are in the same address group).
```

Comment:
This part didn't seem to make much sense originally and now it looks wrong.

In any case, I'd just remove this test, instead of trying to clean it up, because IMO it's confusing and rather useless. In fact, I think we only need the `discovered_node_separate_groups` one (which can be named just `node_discovery`).

Reply:
Isn't it usefull to have tests that show that the number of connection is different if some nodes belong to the same group?

Reply:
Well, in general it does seem useful, but it also looks redundant in these tests, whose purpose is to check address discovery. On the other hand, we don't seem to have dedicated tests that would check this address-group-related logic.
If you decide to keep it, plz rephrase the comment a little, so that it's consistent with the similar comments in other tests.

Reply:
I'll leave them and update the comment